### PR TITLE
Convert MessageModalComponent - AB#15793

### PR DESCRIPTION
**Pull request by BronzBierd**
# Implements [AB#15793](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15793)

## Description

1. Requires [AB#15790](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15790)
3. Converted Message modal
    * Added to `shared` folder
4. Reintroduced Snowplow
5. Rewired message modal into home view post [AB#15790](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15790)
6. Added xlAndUp for the final col number on home, as there was a bug on UHD screen which set the width of the cards to the full screen length.

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [X] Not Required

## UI Changes
image

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/WebClient/src/NewClientApp/src/components/shared/MessageModalComponent.vue [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        persistent
+        no-click-animation
+    >
+        <div class="d-flex justify-center">
+            <v-card :loading="isLoading" max-width="700px">
+                <template v-slot:loader="{ isActive }">
+                    <v-progress-linear
+                        :active="isActive"
+                        color="primary"
+                        heigh="4"
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/components/shared/MessageModalComponent.vue:58]:
```suggestion
```
Typo here, but the default height is already 4.

File: Apps/WebClient/src/NewClientApp/src/components/shared/MessageModalComponent.vue [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        v-model="isVisible"
+        persistent
+        no-click-animation
+    >
+        <div class="d-flex justify-center">
+            <v-card :loading="isLoading" max-width="700px">
+                <template v-slot:loader="{ isActive }">
+                    <v-progress-linear
+                        :active="isActive"
+                        color="primary"
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/components/shared/MessageModalComponent.vue:57]:
```suggestion
                        color="accent"
```
A primary-coloured loading indicator will not be visible against the background of the title bar (also primary colour).

File: Apps/WebClient/src/NewClientApp/src/components/shared/MessageModalComponent.vue [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                        :variant="okOnly ? 'primary' : 'secondary'"
+                        @click.prevent="handleCancel"
+                        :text="okOnly ? 'OK' : 'Cancel'"
+                    />
+                    <HgButtonComponent
+                        data-testid="genericMessageSubmitBtn"
+                        variant="primary"
+                        @click.prevent="handleSubmit"
+                        text="Continue"
+                    />
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/components/shared/MessageModalComponent.vue:89]:
```suggestion
                    <HgButtonComponent
                        data-testid="genericMessageOkBtn"
                        :variant="okOnly ? 'primary' : 'secondary'"
                        :text="okOnly ? 'OK' : 'Cancel'"
                        @click.prevent="handleCancel"
                    />
                    <HgButtonComponent
                        v-if="!okOnly"
                        data-testid="genericMessageSubmitBtn"
                        class="ml-2"
                        variant="primary"
                        text="Continue"
                        @click.prevent="handleSubmit"
                    />
```

Comment (MikeLyttle) [reply Apps/WebClient/src/NewClientApp/src/components/shared/MessageModalComponent.vue:89]: The Continue button should not appear when okOnly is true.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

**Review by BrianMaki — APPROVED** [review]

**Review by MikeLyttle — APPROVED** [review]

**Pull request merged**